content/docs/userspace/gall-guide/3-imports-and-aliases.md

```diff
@@ -0,0 +1,292 @@
++++
+title = "3. Imports and Aliases"
+weight = 15
+template = "doc.html"
++++
+
+In the last lesson we looked at the most basic aspects of a Gall agent's
+structure. Before we get into the different agent arms in detail, there's some
+boilerplate to cover that makes life easier when writing Gall agents.
+
+## Useful libraries
+
+There are a couple of libraries that you'll very likely use in every agent you
+write. These are [`default-agent`](#default-agent) and [`dbug`](#dbug). In
+brief, `default-agent` provides simple default behaviours for each agent arm,
+and `dbug` lets you inspect the state and bowl of an agent from the dojo, for
+debugging purposes. Every example agent we look at from here on out will make
+use of both libraries.
+
+Let's look at each in more detail:
+
+### `default-agent`
+
+The `default-agent` library contains a basic agent with sane default behaviours
+for each arm. In some cases it just crashes and prints an error message to the
+terminal, and in others it succeeds but does nothing. It has two primary uses:
+
+- For any agent arms you don't need, you can just have them call the matching
+  function in `default-agent`, rather than having to manually handle events on
+  those arms.
+- A common pattern in an agent is to switch on the input of an arm with
+  [wutlus](/docs/hoon/reference/rune/wut#-wutlus) (`?+`) runes or maybe
+  [wutcol](/docs/hoon/reference/rune/wut#wutcol) (`?:`) runes. For any
+  unexpected input, you can just pass it to the relevant arm of `default-agent`
+  rather than handling it manually.
+
+The `default-agent` library lives in `/lib/default-agent/hoon` of the `%base`
+desk, and you would typically include a copy in any new desk you created. It's
+imported at the beginning of an agent with the
+[faslus](/docs/arvo/ford/ford#ford-runes) (`/+`) rune.
+
+The library is a wet gate which takes two arguments: `agent` and `help`. The
+first is your agent core itself, and the second is a `?`. If `help` is `%.y`, it
```

Comment:
```suggestion
first is your agent core itself, and the second is a `?`. If `help` is `%.y` (equivalently, `%&`), it
```
important to mention this other convention since it appears frequently